docx2txt/docx_file.py

```diff
@@ -0,0 +1,251 @@
+    """Write image data to disk
+
+    Arguments:
+        img_bytes {bytes} -- image data
+        img_dir {str} -- output directory
+        fname {str} -- name of source file
+
+    Returns:
+        str -- absolute path to extracted image
+    """
+    dst_fname = os.path.join(img_dir, os.path.basename(fname))
+
+    with open(dst_fname, 'wb') as dst_f:
+        dst_f.write(img_bytes)
+
+    return os.path.abspath(dst_fname)
+
+
+def qn(tag):
+    """
+    Stands for 'qualified name', a utility function to turn a namespace
+    prefixed tag name into a Clark-notation qualified tag name for lxml. For
+    example, ``qn('p:cSld')`` returns ``'{http://schemas.../main}cSld'``.
+    Source: https://github.com/python-openxml/python-docx/
+    """
+    nsmap = {
+        'w': 'http://schemas.openxmlformats.org/wordprocessingml/2006/main'}
+    prefix, tagroot = tag.split(':')
+    uri = nsmap[prefix]
+    return '{{{}}}{}'.format(uri, tagroot)
+
+
+def un_qn(tag):
+    # type: (str) -> str
+    """Stands for 'unqualified name'. Removes namespace from prefixed tag.
+
+    See: [Python issue 18304](https://bugs.python.org/issue18304)
+
+    Arguments:
+        tag {str} -- (possibly-)namespaced tag
+
+    Returns:
+        str -- tag name without namespace
+    """
+    return tag.split('}').pop()
+
+
+def xml2text(xml):
+    """
+    A string representing the textual content of this run, with content
+    child elements like ``<w:tab/>`` translated to their Python
+    equivalent.
+    Adapted from: https://github.com/python-openxml/python-docx/
+    """
+    text = u''
+    root = ET.fromstring(xml)
+    whitespace_tags = {
+        qn('w:tab'): '\t',
+        qn('w:br'): '\n',
+        qn('w:cr'): '\n',
+        qn('w:p'): '\n\n', }
+    text_tag = qn('w:t')
+    for child in root.iter():
+        text += whitespace_tags.get(child.tag, '')
+        if child.tag == text_tag and child.text is not None:
+            text += child.text
+    return text
+
+
+def xml2dict(xml):
+    # type: (bytes) -> dict
+    """Get dictionary of values from ``xml``
+
+    Arguments:
+        xml {bytes} -- contents of XML file
+
+    Returns:
+        dict -- dictionary of {node.tagName: node.text}
+    """
+    root = ET.fromstring(xml)
+    data = {
+        un_qn(child.tag): child.text
+        for child in root.iter()}
+    return data
+
+
+def parse_docx(path, img_dir):
+    # type: (str, str) -> dict
+    """Load and parse contents of file at ``path``
+
+    Arguments:
+        path {str} -- path to DOCX file
+
+    Keyword Arguments:
+        img_dir {str} -- save images in specififed directory (default: {None})
+
+    Returns:
+        dict -- header, main, footer, images, and properties of DOCX file
+    """
+    TEXT_KEYS = ['header', 'officeDocument', 'footer']
+    PROP_KEY = 'properties'
+    IMG_KEY = 'image'
+
+    zipf = zipfile.ZipFile(path)
+    paths = {}
+    for fname in ['_rels/.rels', 'word/_rels/document.xml.rels']:
```

Comment:
at least in my office365 docx, the second file is named `document2.xml.rels` so it might not be a good idea to hardcode this, but instead deduce its name from the content of the `_rels/.rels` file.

Reply:
If it helps, here you have a simple Word Online document.

[SimpleDocument.zip](https://github.com/ankushshah89/python-docx2txt/files/2844892/SimpleDocument.zip)

Reply:
It was extremely helpful. Looks like 365 does this for no particular reason.